File: iosApp/iosApp/Theme/AppColors.swift
import SwiftUI
import Shared

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: .alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let r, g, b: UInt64
        switch hex.count {
        case 6: (r, g, b) = ((int >> 16) & 0xFF, (int >> 8) & 0xFF, int & 0xFF)
        default: (r, g, b) = (0, 0, 0)
        }
        self.init(red: Double(r) / 255, green: Double(g) / 255, blue: Double(b) / 255)
    }
}

private let t = AppColorTokens.shared

struct AppColors {
    let background: Color
    let card: Color
    let cardBorder: Color
    let surface: Color
    let mutedText: Color
    let primaryText: Color
    let navBar: Color
    let headerBg: Color
    let rowAltBg: Color
    let isDark: Bool

    static let racingRed = Color(hex: t.racingRed)
    static let f1Red     = Color(hex: t.f1Red)

    static func forScheme(_ scheme: ColorScheme) -> AppColors {
        switch scheme {
        case .dark:
            return AppColors(
                background: Color(hex: t.darkBackground),
                card:        Color(hex: t.darkCard),
                cardBorder:  Color(hex: t.darkCardBorder),
                surface:     Color(hex: t.darkSurface),
                mutedText:   Color(hex: t.darkMutedText),
                primaryText: .white,
                navBar:      Color(hex: t.darkNavBar),
                headerBg:    Color(hex: t.darkHeaderBg),
                rowAltBg:    Color(hex: t.darkRowAltBg),
                isDark: true
            )
        default:
            return AppColors(
                background: Color(hex: t.lightBackground),
                card:        Color(hex: t.lightCard),
                cardBorder:  Color(hex: t.lightCardBorder),
                surface:     Color(hex: t.lightSurface),
                mutedText:   Color(hex: t.lightMutedText),
                primaryText: .black,
                navBar:      Color(hex: t.lightNavBar),
                headerBg:    Color(hex: t.lightHeaderBg),
                rowAltBg:    Color(hex: t.lightRowAltBg),
                isDark: false
            )
        }
    }

    // MARK: - Team colors (sourced from AppColorTokens)
    static var teamColor: (String) -> Color = { team in
        let lower = team.lowercased()
        if lower.contains("mercedes") { return Color(hex: t.teamMercedes) }
        if lower.contains("mclaren")  { return Color(hex: t.teamMcLaren)  }
        if lower.contains("red bull") { return Color(hex: t.teamRedBull)  }
        if lower.contains("ferrari")  { return Color(hex: t.teamFerrari)  }
        if lower.contains("aston")    { return Color(hex: t.teamAston)    }
        if lower.contains("alpine")   { return Color(hex: t.teamAlpine)   }
        if lower.contains("williams") { return Color(hex: t.teamWilliams) }
        if lower.contains("rb")       { return Color(hex: t.teamRb)       }
        if lower.contains("haas")     { return Color(hex: t.teamHaas)     }
        if lower.contains("sauber")   { return Color(hex: t.teamSauber)   }
        return Color(hex: t.teamDefault)
    }

    // MARK: - Podium accents (sourced from AppColorTokens)
    static func podiumAccent(position: Int, fallback: Color) -> Color {
        switch position {
        case 1:  return Color(hex: t.gold)
        case 2:  return Color(hex: t.silver)
        case 3:  return Color(hex: t.bronze)
        default: return fallback
        }
    }
}

File: composeApp/src/androidMain/kotlin/org/gce/racehub/theme/AppColors.kt
package org.gce.racehub.theme

import androidx.compose.runtime.compositionLocalOf
import androidx.compose.ui.graphics.Color

fun hexColor(token: String): Color = Color(0xFF000000L or token.toLong(16))

val RacingRed = hexColor(AppColorTokens.racingRed)

data class AppColorScheme(
    val background: Color,
    val card: Color,
    val cardBorder: Color,
    val surface: Color,
    val mutedText: Color,
    val primaryText: Color,
    val headerBg: Color,
    val rowAltBg: Color,
    val navBar: Color,
    val isDark: Boolean
) {
    val racingRed: Color get() = RacingRed
}

val DarkAppColors = AppColorScheme(
    background = hexColor(AppColorTokens.darkBackground),
    card = hexColor(AppColorTokens.darkCard),
    cardBorder = hexColor(AppColorTokens.darkCardBorder),
    surface = hexColor(AppColorTokens.darkSurface),
    mutedText = hexColor(AppColorTokens.darkMutedText),
    primaryText = Color.White,
    headerBg = hexColor(AppColorTokens.darkHeaderBg),
    rowAltBg = hexColor(AppColorTokens.darkRowAltBg),
    navBar = hexColor(AppColorTokens.darkNavBar),
    isDark = true
)

val LightAppColors = AppColorScheme(
    background = hexColor(AppColorTokens.lightBackground),
    card = hexColor(AppColorTokens.lightCard),
    cardBorder = hexColor(AppColorTokens.lightCardBorder),
    surface = hexColor(AppColorTokens.lightSurface),
    mutedText = hexColor(AppColorTokens.lightMutedText),
    primaryText = Color.Black,
    headerBg = hexColor(AppColorTokens.lightHeaderBg),
    rowAltBg = hexColor(AppColorTokens.lightRowAltBg),
    navBar = hexColor(AppColorTokens.lightNavBar),
    isDark = false
)

val LocalAppColors = compositionLocalOf { DarkAppColors }

fun teamColorOf(team: String): Color {
    val t = team.lowercase()
    return when {
        "mercedes" in t -> hexColor(AppColorTokens.teamMercedes)
        "mclaren" in t  -> hexColor(AppColorTokens.teamMcLaren)
        "red bull" in t -> hexColor(AppColorTokens.teamRedBull)
        "ferrari" in t  -> hexColor(AppColorTokens.teamFerrari)
        "aston" in t    -> hexColor(AppColorTokens.teamAston)
        "alpine" in t   -> hexColor(AppColorTokens.teamAlpine)
        "williams" in t -> hexColor(AppColorTokens.teamWilliams)
        "rb" in t || "racing bulls" in t -> hexColor(AppColorTokens.teamRb)
        "haas" in t     -> hexColor(AppColorTokens.teamHaas)
        "sauber" in t || "kick" in t -> hexColor(AppColorTokens.teamSauber)
        else            -> hexColor(AppColorTokens.teamDefault)
    }
}
